eng/common/scripts/TypeSpec-Project-Sync.ps1

```diff
@@ -93,7 +97,15 @@ $configuration = Get-Content -Path $typespecConfigurationFile -Raw | ConvertFrom
 $pieces = $typespecConfigurationFile.Path.Replace("\","/").Split("/")
 $projectName = $pieces[$pieces.Count - 2]
 
-$specSubDirectory = $configuration["directory"]
+# clone the whole RP directory which is the parent of $configuration["directory"]
+if ($configuration["directory"] -match "^[^/\\]+[\\/]+[^/\\]+")
```

Comment:
What kind of pattern is this supposed to match?

Reply:
Should we just walk up the parents until we hit the "specification" folder?

Reply:
Yes. It will get the top-level directory under `specification`.

Reply:
This regex doesn't say to me that it is looking for specification folder. What do we expect the values of `$configuraion["directory"]` to be? Are they relative paths from the specification folder? or from the root of the repo? If they are from the root I would expect this regex to have the work "specification" in it somewhere.